irksome/imex.py

```diff
@@ -304,3 +307,225 @@ def solver_stats(self):
                 self.num_linear_iterations_prop,
                 self.num_nonlinear_iterations_it,
                 self.num_linear_iterations_it)
+
+
+def getFormsIMEX(F, Fexp, ks, khats, butch, t, dt, u0, bcs=None):
+    if bcs is None:
+        bcs = []
+
+    v = F.arguments()[0]
+    V = v.function_space()
+    msh = V.mesh()
+    assert V == u0.function_space()
+
+    num_fields = len(V)
+    num_stages = butch.num_stages
+    k = Function(V)
+    g = Function(V)
+
+    khat = Function(V)
+    ghat = Function(V)
+    vhat = TestFunction(V)
+
+    # If we're on a mixed problem, we need to replace pieces of the
+    # solution.  Stores array of the splittings of the functions for each stage.
+    if num_fields == 1:
```

Comment:
sometimes these are lists, sometimes `np.array`?  What's up with that?

Reply:
Good question (and one that I asked myself).  This is copied/extended from [here](https://github.com/firedrakeproject/Irksome/blob/cbf6af426277fd84369c3fa6cdea211305d004be/irksome/dirk_stepper.py#L29).  It ain't broke, but I'm not opposed to fixing it.